main/modes/test/swadgePassTest/swadgePassTest.c

```diff
@@ -0,0 +1,208 @@
+//==============================================================================
+// Includes
+//==============================================================================
+
+#include "swadgePassTest.h"
+
+//==============================================================================
+// Structs
+//==============================================================================
+
+/// @brief The struct that holds all the state for the SwadgePass test mode
+typedef struct
+{
+    menu_t* menu;
+    menuManiaRenderer_t* renderer;
+    list_t swadgePasses;
+    swadgePassData_t* currSpd;
+} swadgePassTest_t;
+
+//==============================================================================
+// Function Prototypes
+//==============================================================================
+
+static void swadgePassTestMainLoop(int64_t elapsedUs);
+static void swadgePassTestEnterMode(void);
+static void swadgePassTestExitMode(void);
+void swadgePassTestMenuCb(const char* label, bool selected, uint32_t value);
+
+//==============================================================================
+// Strings
+//==============================================================================
+
+static const char swadgePassTestName[] = "SwadgePass Test";
+
+//==============================================================================
+// Variables
+//==============================================================================
+
+/// The Swadge mode for swadgePassTest
+swadgeMode_t swadgePassTestMode = {
+    .modeName                 = swadgePassTestName,
+    .wifiMode                 = NO_WIFI,
+    .overrideUsb              = false,
+    .usesAccelerometer        = false,
+    .usesThermometer          = false,
+    .overrideSelectBtn        = false,
+    .fnEnterMode              = swadgePassTestEnterMode,
+    .fnExitMode               = swadgePassTestExitMode,
+    .fnMainLoop               = swadgePassTestMainLoop,
+    .fnAudioCallback          = NULL,
+    .fnBackgroundDrawCallback = NULL,
+    .fnEspNowRecvCb           = NULL,
+    .fnEspNowSendCb           = NULL,
+    .fnAdvancedUSB            = NULL,
+};
+
+/// All state information for the SwadgePass Test mode.
+swadgePassTest_t* spt = NULL;
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+/**
+ * @brief Enter SwadgePass Test mode, allocate required memory, and initialize required variables
+ *
+ */
+static void swadgePassTestEnterMode(void)
+{
+    spt = heap_caps_calloc(1, sizeof(swadgePassTest_t), MALLOC_CAP_8BIT);
+
+    // Get SwadgePasses
+    getSwadgePasses(&spt->swadgePasses, &swadgePassTestMode, true);
+
+    // Initialize menu
+    spt->menu = initMenu(swadgePassTestName, swadgePassTestMenuCb);
+
+    // Add all keys to the menu
+    node_t* passNode = spt->swadgePasses.first;
+    while (passNode)
```

Comment:
Not a huge issue for the test mode but if there are no swadge passes at all, the empty menu seems to cause a crash after a few seconds.

Reply:
Hrm, I'm not replicating that with the emulator (deleted `nvs.json`) or firmware (did a factory reset from the secret menu). Maybe you had stale data or something?